### ISSUE TITLE
buttons in entities card doesn't support same functions as button card

### ISSUE THREAD

**Issue by scyto** [home-assistant/frontend#8188]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## Steps to reproduce

1. prove out basic functionality via the button card.  Here is example 
```
  - type: horizontal-stack
    cards:
      - type: button
        icon: 'mdi:volume-plus'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            entity_id: remote.basement_harmony
            command: VolumeUp
            device: 49625216
        icon_height: 45px
      - type: button
        icon: 'mdi:volume-off'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            entity_id: remote.basement_harmony
            command: Mute
            device: 49625216
        icon_height: 45px
      - type: button
        icon: 'mdi:volume-minus'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            entity_id: remote.basement_harmony
            command: VolumeDown
            device: 49625216
        icon_height: 45px
```
2 try and reproduce these buttons in the entities card.  I tried two variants:
```
type: entities
entities:
  type: entities
entities:
  - type: divider
  - type: buttons
    entities:
      - type: button
        icon: 'mdi:volume-plus'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            entity_id: remote.basement_harmony
            command: VolumeUp
            device: 9625216
      - type: button
        icon: 'mdi:volume-minus'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            entity_id: remote.basement_harmony
            command: VolumeDown
            device: 9625216
  - type: divider
```
and
```
type: entities
entities:
  - type: divider
  - type: buttons
    entities:
      - type: button
        icon: 'mdi:volume-plus'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            command: VolumeUp
            entity_id: 9625216
      - type: button
        icon: 'mdi:volume-minus'
        tap_action:
          action: call-service
          service: remote.send_command
          service_data:
            command: VolumeDown
            entity_id: 9625216
  - type: divider
```
I am not assuming that these should be supported not am I assuming that my yaml is right.  If there are edits to my code that would make this work that is awesome.  I couldn't figure it out from the docs (i am new).

## Environment

- Home Assistant release with the issue: 2021.1.4
- Last working Home Assistant release (if known): never worked AFAIK
- Browser and browser version: Latest new windows Edge browser.
- Operating system: Windows 10

## State of relevant entities

saved cards look like this
![image](https://user-images.githubusercontent.com/11369180/104856128-a41b8280-58c5-11eb-95be-97456b0f2b4a.png)
and 
![image](https://user-images.githubusercontent.com/11369180/104856138-b09fdb00-58c5-11eb-9ea1-966ccabac005.png)

```
none

```

## Problem-relevant frontend configuration

```
provided in repro above

```

## Javascript errors shown in your browser console/inspector

```
none

```

## System Health

version | 2021.1.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
virtualenv | false
python_version | 3.8.7
os_name | Linux
os_version | 5.4.86
arch | x86_64
timezone | America/Los_Angeles

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4994
Installed Version | 1.10.0
Stage | running
Available Repositories | 709
Installed Repositories | 2

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Hass.io</summary>

host_os | Home Assistant OS 5.10
-- | --
update_channel | stable
supervisor_version | 2021.01.4
docker_version | 19.03.13
disk_total | 30.8 GB
disk_used | 6.3 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Mosquitto broker (5.1), Samba share (9.3.0), Visual Studio Code (2.9.1), Log Viewer (0.9.1), SSH & Web Terminal (7.8.0)

</details>

<details><summary>Lovelace</summary>

dashboards | 1
-- | --
mode | storage
views | 6
resources | 1

</details>

**Comment by spacegaier** [comment]
Your entity IDs in the service payload are not valid. They should be something in the format `"<domain>.<id>"`. Also the `entities` list of `buttons` requires an entity ID each (`entity`).

**Comment by scyto** [comment]
oops, let me correct that, pretty certain I had that last night

**Comment by scyto** [comment]
I corrected both examples.  
No major change (see shots below)
What i find confusing is that in the intgration the data is called data but in the card it is called service_data - not sure if that matters, but differences in syntax sure are confusing?

![image](https://user-images.githubusercontent.com/11369180/104863059-d2ab5480-58e9-11eb-871d-61a90e2185e1.png)
 and
![image](https://user-images.githubusercontent.com/11369180/104863077-dccd5300-58e9-11eb-9093-12187527493d.png)

this is the integrations doc https://www.home-assistant.io/integrations/harmony/#service-remotesend_command

**Comment by iantrich** [comment]
In a buttons row, the entities require entities as already mentioned. Your update does not appear to address that still. You also do not need to specify type in the entities within a buttons row.

**Comment by spacegaier** [comment]
> 
> 
> this is the integrations doc https://www.home-assistant.io/integrations/harmony/#service-remotesend_command

That is not a frontend example, but the backend variant, e.g. used in an automation or a script. In those instances it is clear that the data belongs to the service (so `data` is clear enough). But in the frontend scenario we are talking about here, the original developer probably wanted to make it clear that this data only relates to the service and is not relevant e.g. for `navigate` or `url` hence `service_data`.

**Comment by scyto** [comment]
@iantrich @spacegaier 

I am sorry I am not sure I understand what you are trying to tell me.

In the entities card it defines entities as 'A list of entity IDs or entity objects or special row objects'

Are you saying that this definition of entities  doesn't apply why it is the entities section of the buttons special row object (or any special row) - aka a button can only support entity IDs or entity objects'.

If so as a newcomer I would have to observe that this is a syntactical and logical nightmare  that the same word does not mean the same thing across home assistant....

I guess then the ask is that 'buttons' supports the other 'special row types'  - i would also advocate there needs to be a lot more consistency across the built-in cards - because decoding this is not easy.

**Comment by scyto** [comment]
> > this is the integrations doc https://www.home-assistant.io/integrations/harmony/#service-remotesend_command
> 
> That is not a frontend example, but the backend variant, e.g. used in an automation or a script. In those instances it is clear that the data belongs to the service (so `data` is clear enough). But in the frontend scenario we are talking about here, the original developer probably wanted to make it clear that this data only relates to the service and is not relevant e.g. for `navigate` or `url` hence `service_data`.

Thanks, you can see why i was concerned if data and service_data was the same thing.  I agree on your supposition of the original intent of the developer, i also get things evolve organically.

**Comment by spacegaier** [comment]
> 
>  
> Are you saying that this definition of entities doesn't apply when it is the entities section of the buttons special row object (or any special row) - aka a button can only support entity IDs or entity objects but not other special rows?'.

Yes, those are different blocks in the YAML. The `entities` list of the card can contain either an entity ID an entity object or a special row definition. What each special definition allows is then listed in turn further down on the page. That also means the `entities` list of the card is not the same as the `entities` list of a `buttons` row.

>
> 
> I guess then the ask is that 'buttons' supports the other 'special row types' - i would also advocate there needs to be a lot more consistency across the built-in cards - because decoding this is not easy.

I don't get that. How can a `buttons` row "support" a `divider` row? Those are separate rows and have nothing to do with each other.

I can see how some parts can be a bit confusing. In the end the naming is something that evolved over time. Renaming things now would break everyone dashboards. Many things are also already possible through the card editors, so people need to do less and less in YAML directly.

I already had some changes merged into the page last night (e.g. more extensive example at the end of the page now also with a screenshot) and more changes are currently in review.

**Comment by scyto** [comment]
@spacegaier thanks, i didn't mean to imply it should be renamed, just explain the root of my confusion.
I just refreshed the page, i like the changes! (esp the expanded conditional example). Thanks.

At this point it sounds like the special row type 'buttons' isn't designed to do what I want, so I got an answer.

Sounds like the idea of the 'buttons' special row being able to include a subset of the other special rows (aka things we migt want buttons to do [i agree divider makes no sense, unless it was vertical, in buttons ;-)] ) is probably best served by a discussion not an issue?

as such closing, thanks for being patient.
